From 9786555acc4bf8348938c55d29c59562d1db4367 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:23:48 -0400
Subject: [cpufreq] intel_pstate: Use del_timer_sync in intel_pstate_cpu_stop

Message-id: <1409235879-1283-135-git-send-email-prarit@redhat.com>
Patchwork-id: 89472
O-Subject: [RHEL7.1 PATCH BZ 1134639 134/185] intel_pstate: Use del_timer_sync in intel_pstate_cpu_stop
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit c2294a2f7853e6450361d078b65407bdaa6d1d11
Author: Dirk Brandewie <dirk.j.brandewie@intel.com>
Date:   Mon Mar 24 07:41:29 2014 -0700

    intel_pstate: Use del_timer_sync in intel_pstate_cpu_stop

    Ensure that no timer callback is running since we are about to free
    the timer structure.  We cannot guarantee that the call back is called
    on the CPU where the timer is running.

    Reported-by: Thomas Gleixner <tglx@linutronix.de>
    Signed-off-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
    Reviewed-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    Acked-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index 3a5cc92..1d777cf 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -781,7 +781,7 @@ static void intel_pstate_stop_cpu(struct cpufreq_policy *policy)
 
  pr_info("intel_pstate CPU %d exiting\n", cpu_num);
 
- del_timer(&all_cpu_data[cpu_num]->timer);
+ del_timer_sync(&all_cpu_data[cpu_num]->timer);
  intel_pstate_set_pstate(cpu, cpu->pstate.min_pstate);
  kfree(all_cpu_data[cpu_num]);
  all_cpu_data[cpu_num] = NULL;
-- 
1.7.1